### PR TITLE
Update inline notice component

### LINKED ISSUE #7048: Update BannerNotice and InlineNotice components
### Description
As part of the new onboarding project, we need to use the banner notice component from WC Design, which is not available to import. So we need to implement it in WooPayments.

We already have `BannerNotice` and `InlineNotice` components, but they are mixed and don't follow our design guidelines.

### Acceptance criteria
- Update `BannerNotice` and `InlineNotice` components following design specs.
- Check every `BannerNotice` and `InlineNotice` instance and be sure the right one is in use.

### Designs
| BannerNotice | InlineNotice |
|-|-|
|Screenshot 2023-08-23 at 09 39 53|Screenshot 2023-08-23 at 10 41 04|

### Additional context
Figma: Kofpo8jj4zDWXDsVdlSz0f-fi-4171_46004

### PR DESCRIPTION
**Pull request by ismaeldcom**
Partially fixes #7048

#### Changes proposed in this Pull Request
Following WC design guidelines (Kofpo8jj4zDWXDsVdlSz0f-fi-4172_49449) we should use banner notices at the top of admin page, and inline ones for any other cases. Our `InlineNotice` component is actually the WP notice one without margins, which matches the banner one.

- Replace all `InlineNotice` instances with `BannerNotice`.
- Remove `InlineNotice` component.
- Update `BannerNotice` component to use icons based on its status, avoiding the need to explicitly declare an icon every time we use the component. But allowing us to use a custom one if required.
- Update every `BannerNotice` instance to use default icons whether possible.
- Rename `BannerNotice` to `InlineNotice`.

I'm splitting the issue in two PRs to make it easier to review, due to changes in too many files. Covering inline notice component in this one, and banner notice in the next one.

#### Testing instructions
- Code changes make sense.
- There are no UI issues like big margins, or displaced elements.
 
The replaced notices are those on the first commit, as shown on the following screenshots.

#### Old vs new notices
enable-upe
feedback
disable-upe
risk-level

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7078` or branch name `update/7048-inline-notice-component` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=update/7048-inline-notice-component) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: caf524905bcf0e561ddf45482c2501d94e128556
- Build time: 2023-08-28 09:37:32 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Comment by ismaeldcom** [comment]
Hey @elizaan36! Could you do a quick review to confirm if this is the intended outcome of #7048 and WooPayments notices in general? 🙂

**Comment by elizaan36** [comment]
@ismaeldcom LGTM! Wonderful update!

File: client/components/inline-notice/index.tsx [hunk 1]
```diff
@@ -1,23 +1,111 @@
...
+
+	// Use default icon based on status if icon === true.
+	let iconToDisplay = icon;
+	if ( iconToDisplay === true ) {
+		switch ( noticeProps.status ) {
+			case 'success':
+				iconToDisplay = <CheckmarkIcon />;
+				break;
+			case 'error':
+			case 'warning':
```
Comment (oaratovskyi) [inline client/components/inline-notice/index.tsx:47]: I am curious if it makes sense to have different icons for error and warning?

**Review by oaratovskyi — APPROVED** [review]
Great work @ismaeldcom ! 
Tested all screens and reviewed the code 🚀 
Special thanks for fixing margin issues in fraud protection and disable UPE/provide feedback 👍

File: client/components/inline-notice/index.tsx (hunk 1 above)
Comment (ismaeldcom) [reply client/components/inline-notice/index.tsx:47]: I thought about it too, but according to our design guidelines (Kofpo8jj4zDWXDsVdlSz0f-fi-4172_49449) they share the same icon 🙂

**Pull request merged**